lasagne/layers/helper.py

```diff
@@ -503,5 +503,10 @@ def set_all_param_values(layer, values, **tags):
     if len(params) != len(values):
         raise ValueError("mismatch: got %d values to set %d parameters" %
                          (len(values), len(params)))
+
     for p, v in zip(params, values):
-        p.set_value(v)
+        if p.get_value().shape != v.shape:
+            raise ValueError("mismatch: parameter has shape %r but value to set \
+                              has shape %r" % (p.get_value().shape, v.shape))
```

Comment:
It's probably better to do this instead:

```
raise ValueError("mismatch: parameter has shape %r but value to set "
                 "has shape %r" % (p.get_value().shape, v.shape))
```

Python will concatenate adjacent strings. Otherwise won't the string have a bunch of spaces in it because of the indentation? Either way, that's how we do it elsewhere in the code so if it doesn't make a difference, it would be nice to change it just for consistency's sake.